requeue/requeue.go

```diff
@@ -65,28 +64,28 @@ func (r *RequeueSubsystem) requeueCommand(c *server.Connection, s *server.Server
 		return
 	}
 
-	q, err := s.Store().GetQueue(job.Queue)
+	q, err := s.Store().GetQueue(c.Context, job.Queue)
 	if err != nil {
 		_ = c.Error(cmd, err)
 		return
 	}
 
 	pushChain := pushChain(s.Manager())
-	ctx := newMiddlewareCtx(context.Background(), job, s.Manager())
 	// Our final function in the middleware chain is based off of the
 	// manager.Push -> manager.enqueue -> redisQueue.Push implementation upstream.
-	// https://github.com/contribsys/faktory/blob/v1.6.2/manager/manager.go#L233
-	// https://github.com/contribsys/faktory/blob/v1.6.2/manager/manager.go#L254
-	// https://github.com/contribsys/faktory/blob/v1.6.2/storage/queue_redis.go#L104
-	err = callMiddleware(pushChain, ctx, func() error {
+	// https://github.com/contribsys/faktory/blob/v1.8.0/manager/manager.go#L222-L236
+	// https://github.com/contribsys/faktory/blob/v1.8.0/manager/manager.go#L251-L255
+	// https://github.com/contribsys/faktory/blob/v1.8.0/storage/queue_redis.go#L104
+	ctxh := context.WithValue(c.Context, manager.MiddlewareHelperKey, Ctx{job, s.Manager(), nil})
```

Comment:
Why is it necessary to add this context value? Why is it okay to now pass our own `Ctx` implementation rather than a `manager.Ctx`?

Reply:
The `requeue` plugin is slightly different from the others because rather than solely installing a middleware this plugin is adding a new command to the wire protocol.

When someone sends a `REQUEUE <jid>` command over the wire, this function is responsible for finding the job in memory and kicking off the middleware chain. The middlewares expect that they will be able to pull a `manager.Context` value from `ctx`, so this function is responsible for handling that. If you follow the links in the comments above this line you’ll see that this is implemented very similarly in the function that handles the `PUSH …` command.

As for why it’s acceptable to use our own `Ctx`, that’s because the middlewares expect a value that implements the `manager.Context` interface and the `manager.Ctx` struct only has unexported fields which make it impossible to construct one outside of the `manager` package.